js/tuxexchange.js

```diff
@@ -0,0 +1,533 @@
+        };
+        const orderBook = await this.publicGet (this.extend (orderBookRequest, params));
+        const result = this.parseOrderBook (orderBook);
+        return result;
+    }
+
+    parseTrade (trade, market) {
+        // The API exposees different fields for the gettrades and getmytradehistory endpoints
+        const isPrivateTrade = trade['orderId'] !== undefined;
+        const tradeDate = this.parseDate (this.safeString (trade, 'date'));
+        const timestamp = tradeDate.valueOf ();
+        let priceString = undefined;
+        if (isPrivateTrade) {
+            priceString = this.safeString (trade, 'price');
+        } else {
+            priceString = this.safeString (trade, 'rate');
+        }
+        const parsedTrade = {
+            'id': this.safeString (trade, 'tradeid'),
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market.id,
+            'type': 'limit',
+            'side': this.safeString (trade, 'type'),
+            'price': this.asFloat (priceString),
+            'amount': this.asFloat (this.safeString (trade, 'amount')),
+            'cost': this.asFloat (this.safeString (trade, 'total')),
+        };
+        if (isPrivateTrade) {
+            let feeCurrency = undefined;
+            if (parsedTrade['side'] === 'buy') {
+                feeCurrency = this.commonCurrencyCode (this.safeString (trade, 'market'));
+            } else {
+                feeCurrency = this.commonCurrencyCode (this.safeString (trade, 'coin'));
+            }
+            return this.extend (parsedTrade, {
+                'order': this.safeString (trade, 'order'),
+                'fee': {
+                    'cost': this.safeFloat (trade, 'total'),
+                    'currency': feeCurrency,
+                    'rate': this.safeFloat (trade, 'feepercent'),
+                },
+            });
+        } else {
+            return parsedTrade;
+        }
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTrades () requires a "symbol" argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
+        if (codes['base'] !== 'BTC') {
+            throw new NotSupported (this.id + ' this exchange only trades on symbols with BTC as base');
+        }
+        let tradeHistoryRequest = {
+            'method': 'gettradehistory',
+            'coin': this.currencyId (codes['quote']),
+            'end': this.seconds (),
+        };
+        if (since !== undefined) {
+            const sinceSeconds = since / 1000;
+            this.extend (tradeHistoryRequest, { 'start': sinceSeconds });
+        }
+        const trades = await this.publicGet (this.extend (tradeHistoryRequest, params));
+        const market = this.market (symbol);
```

Comment:
Heh, for some reason, you've used `this.market (symbol)` correctly here, but not on line 240 above: https://github.com/ccxt/ccxt/pull/5176/files#diff-eb1d32ed3d775af83b55a746e0401595R240 )